test/commonControlTests/NumberControl.spec.ts

```diff
@@ -11,608 +11,436 @@
+        await testE2E(requestHandler, [
+            'U: change my age to one hundred and thirteen',
+            TestInput.of(
+                ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, {
+                    'AMAZON.NUMBER': '113',
+                    action: $.Action.Change,
+                }),
+            ),
+            'A: Was that 113?',
+            'U: No',
+            TestInput.of(IntentBuilder.of(AmazonIntent.NoIntent)),
+            'A: My mistake. Did you perhaps mean 130?',
+            'U: no',
+            TestInput.of(IntentBuilder.of(AmazonIntent.NoIntent)),
+            'A: My mistake. How old will you be?',
+            'U: twelve',
+            TestInput.of(
+                ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, {
+                    'AMAZON.NUMBER': '12',
+                }),
+            ),
+            'A: Great.',
+        ]);
+        expect(requestHandler.getSerializableControlStates().ageSelector.value).eq(12);
+    });
 
-        test('handling special number like negative and zero', async () => {
-            const requestHandler = new ControlHandler(new NumberControlManager());
-            await testE2E(requestHandler, [
-                'U: Minus ten.',
-                TestInput.of(ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, { 'AMAZON.NUMBER': '-10' })),
-                "A: Sorry but that's not a valid choice because the value must be between zero and fifty inclusive. How many?",
-                'U: Zero',
-                TestInput.of(ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, { 'AMAZON.NUMBER': '0' })),
-                'A: Was that 0?',
-                'U: Yes',
-                TestInput.of(IntentBuilder.of(AmazonIntent.YesIntent)),
-                'A: Great.',
-            ]);
-            expect(
-                (requestHandler.getSerializableControlStates()[TEST_CONTROL_ID] as NumberControlState).value,
-            ).eq(0);
-        });
+    test('change age value with explicit disaffirmation with values', async () => {
+        const requestHandler = new ControlHandler(new AgeControlManager());
+        await testE2E(requestHandler, [
+            'U: change my age to one hundred and thirteen',
+            TestInput.of(
+                ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, {
+                    'AMAZON.NUMBER': '113',
+                    action: $.Action.Change,
+                }),
+            ),
+            'A: Was that 113?',
+            'U: No',
+            TestInput.of(IntentBuilder.of(AmazonIntent.NoIntent)),
+            'A: My mistake. Did you perhaps mean 130?',
+            'U: no, fifty',
+            TestInput.of(
+                ValueControlIntent.of(AmazonBuiltInSlotType.NUMBER, {
+                    'AMAZON.NUMBER': '50',
+                    feedback: $.Feedback.Disaffirm,
+                }),
+            ),
+            'A: My mistake. Did you perhaps mean 50?',
```

Comment:
hmm this one is surprising..   I would have expected a Confirmation.  "my mistake. was that 50?"

How it it cause a suggestion for the same value the user said?

Reply:
Yes, my bad! Wrong act setup in handleFeedbackWithValue func, fixed it.